pywbem_mock/_wbemconnection_mock.py

```diff
@@ -1132,22 +1125,21 @@ def add_cimobjects(self, objects, namespace=None):
             if isinstance(obj, CIMClass):
                 cc = obj.copy()
                 if cc.superclass:
-                    if not self.mainprovider.class_exists(namespace,
-                                                          cc.superclass):
+                    if not self._mainprovider.class_exists(namespace,
+                                                           cc.superclass):
                         raise ValueError(
                             _format("Class {0!A} defines superclass {1!A} but "
                                     "the superclass does not exist in the "
                                     "repository.",
                                     cc.classname, cc.superclass))
-                class_store = self._get_class_store(namespace)
-
-                qualifier_store = self._get_qualifier_store(namespace)
 
                 # pylint: disable=protected-access
-                cc1 = self.mainprovider._resolve_class(cc, namespace,
-                                                       qualifier_store,
-                                                       verbose=False)
+                cc1 = self._mainprovider._resolve_class(
+                    cc, namespace,
+                    self._mainprovider.get_qualifier_store(namespace),
```

Comment:
To go one step further, I think we should consider also removing the `get_*_store()` convenience methods and the `namespaces` property from `BaseProvider`, since any provider now has access to the repository via its own `cimrepository` property.

Reply:
DONE